src/Query/Mysql/AnyValue.php

```diff
@@ -15,10 +17,12 @@ class AnyValue extends FunctionNode
 
     public function parse(Parser $parser): void
     {
-        $parser->match(Lexer::T_IDENTIFIER);
-        $parser->match(Lexer::T_OPEN_PARENTHESIS);
+        $orm3 = ! defined('\Doctrine\ORM\Query\Lexer::T_IDENTIFIER');
```

Comment:
I would perform a forward feature detection instead.

```suggestion
        $orm3 = class_exsists(TokenType::class);
```